### PR TITLE
Rest client config - turn the config module into a proper extension

### LINKED ISSUE #21488: quarkus.rest-client.* properties not present in all-config guide
### Describe the bug

`quarkus.rest-client.*` properties are not included in https://quarkus.io/guides/all-config. Not sure if this is intended.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TomasHofman**
This is needed for the automatic docs generation to work properly.

Fixes https://github.com/quarkusio/quarkus/issues/21488

### CODE REVIEW

**Comment by geoand** [comment]
Needs a formatting update :)

**Comment by gsmet** [comment]
Hmmm I haven't followed all this but I find it weird we have to introduce an extension just for some config classes.

Could you summarize to me why we ended up with a separate module for the config?

Thanks!

**Comment by TomasHofman** [comment]
Hello @gsmet, so one aspect is that we want to share configuration classes between classic and reactive rest clients - that's why it's in a separate module.

Then we found that the "quarkus.rest-client.*" configuration properties that we define in our config roots are missing from auto-generated docs (https://quarkus.io/guides/all-config). The reason is that the generator class (`AllConfigGenerator`) only detects config roots if they are part of extension jars directly (or that's how it appears to me).

**Review by gsmet — APPROVED** [review]
Looks good to me. Let's merge it to have it for CR1. Thanks for the follow-up!
